csrc/multidevice/lower_communication.cpp

```diff
@@ -11,11 +11,57 @@
 #include <multidevice/lower_communication.h>
 #include <multidevice/pipeline.h>
 #include <multidevice/utils.h>
+#include <ops/all_ops.h>
+#include <limits>
 
 namespace nvfuser {
 
 namespace {
 
+template <typename T>
+inline T getInitialValue(BinaryOpType op) {
+  switch (op) {
+    case BinaryOpType::Add:
+    case BinaryOpType::BitwiseOr:
+    case BinaryOpType::BitwiseXor:
+      return 0;
+    case BinaryOpType::Mul:
+      return 1;
+    case BinaryOpType::Min:
+      return std::numeric_limits<T>::min();
+    case BinaryOpType::Max:
+    case BinaryOpType::BitwiseAnd:
+      return std::numeric_limits<T>::max();
```

Comment:
Here too. Also why BitwiseAnd use `max`? What would type `T` be?

Reply:
Indeed, `BinaryOpType::Max:` needs `std::numeric_limits<T>::min()`

For `BitwiseAnd`, what do you suggest? The neutral element should be 111....1 bitwise, so this is max, right? Let me know if I am wrong

About type T, I think it can be anything for all the ops except BitwiseAnd, for which there will actually be a problem...

Reply:
The best would be to choose T according to the tensor's type, but I couldn't find how to translate torch's type into c++ native type

Reply:
Since `T` for bitwise operations should be an integer type, why not just something like `~(T)0`?

Reply:
> I couldn't find how to translate torch's type into c++ native type

Does `AtenTypeToNativeType` work? See type.h.

Reply:
> Since `T` for bitwise operations should be an integer type, why not just something like `~(T)0`?

Right, I didn't think about that, thx 


> Does AtenTypeToNativeType work? See type.h.

Something like:
`getInitialValue<c10::impl::ScalarTypeToCPPType<reference.scalar_type()>::t>(op_type)`
should work, or alternatively replacing `ScalarTypeToCPPType` with `AtenTypeToNativeType` as you suggest.
But we still have the problem of "non-`constexpr`ness" of the template argument.
We should write a dispatch function or something like that. I don't know what's the best way to handle that. As far as I understand there is no c++ torch call to get the limits from the ScalarType, even though there is a python pytorch way to do it, through finfo.

For now, I just assert that the type is kFloat32 and write a TODO. Lmk if it is ok for you